contxt/models/changetrackingrecord.py

```diff
@@ -0,0 +1,75 @@
+from enum import Enum
+import json
+
+
+class DBOperation(Enum):
+    INSERT = 0
+    UPDATE = 1 
+    UPSERT = 2 
+    DELETE = 3
+
+    @classmethod
+    def has_value(cls, value):
+        return value in cls._value2member_map_ 
+
+class CTRHeader():
+    def __init__(self, forwardable = True, deduplicatable = True):
+        self.forwardable = forwardable
+        self.deduplicatable = deduplicatable
+
+class ChangeTrackingRecord():
+    '''A JSON Serializable ndustrial.io ChangeTrackingRecord(tm) that is
+       ingestable by the ndustrial ETL java/kotlin components.
+       
+       topic: Str           - the name of the canonical being ETL-ed
+       version: Str         - some long identification I'm not sure
+       operation: enum      - a DB operation (insert/update/upsert/delete) from the DBOperation enum
+       source_host: Str     - the hostname/ip/table that homes the source data
+       source_table: Str    - the name of the table from where the data is being extracted
+       source_database: Str - the name of the db being extracted against
+       tuple: dict          - key/value pairs that make up the data payload. This
+                              value defaults to an empty dict and can be updated after creation
+       headers: CTRHeader() - the header information that informs ETL devices what do with the CTR
+    '''
+    def __init__(
+        self,
+        topic: str,
+        version: int,
+        operation: DBOperation,
+        source_host: str,
+        source_table: str,
+        source_database: str,
+        tuple = dict(),
+        headers = CTRHeader()):
+
+        errors = ""
+        if type(topic) is not str:
```

Comment:
For my own python learning: Does python not perform type checking already by declaring the type in the parameters list? If not, what's the benefit of declaring these types up front?

Reply:
For right now at least, Python's type hinting system is only used by linters and static analyzers. 

I figured I'd be a good guy and participate in the system, it at least makes it easier to understand what those should be type-wise at a glance.